src/openid.rs

```diff
@@ -48,6 +52,30 @@ pub fn login(api_endpoint: Url) -> Result<Context> {
     Ok(config)
 }
 
+fn login_with_refresh_token(
+    auth_url: Url,
+    token_url: Url,
+    refresh_token_val: &str,
+) -> Result<BasicTokenResponse> {
+    log::debug!("Using refresh token");
+
+    let auth_url = AuthUrl::new(auth_url.to_string())?;
+    let token_url = TokenUrl::new(token_url.to_string())?;
+
+    let client = BasicClient::new(
+        ClientId::new(CLIENT_ID.to_string()),
+        None,
+        auth_url,
+        Some(token_url.clone()),
+    );
+
+    // Exchange the refresh token for access token
+    client
+        .exchange_refresh_token(&oauth2::RefreshToken::new(refresh_token_val.to_string()))
+        .request(http_client)
+        .map_err(|_| Error::msg("Invalid refresh token"))
+}
+
```

Comment:
We would have two `refresh_token` methods that would do almost the same thing. 
I think you could refactor both to use the same call to `client.exchange_refresh_token()`. 

https://github.com/drogue-iot/drg/blob/3acc5863a734247bb6df263f272a2f5d4ab257b2/src/openid.rs#L126-L155

Reply:
Okay good advice, I have moved the repeated code to a new function `exchange_token`